src/Bicep.Core/Workspaces/SourceFileGroupingBuilder.cs

```diff
@@ -59,7 +59,7 @@ public static SourceFileGrouping Rebuild(IModuleDispatcher moduleDispatcher, IRe
         {
             var builder = new SourceFileGroupingBuilder(current.FileResolver, moduleDispatcher, workspace, current);
             var isParamsFile = current.FileResultByUri[current.EntryFileUri].File is BicepParamFile;
-            var modulesToRestore = current.GetModulesToRestore().ToHashSet();
+            var modulesToRestore = current.GetModulesToRestore().OfType<ModuleSourceResolutionInfo>().ToHashSet();
```

Comment:
>OfType [](http://example.com/codeflow?start=65&length=6)

Method name suggests it returns just modules. Why do we have to filter?

Reply:
we don't need to anymore, it was a crutch to cast the types to the `ModuleSourceResolutionInfo` since the enumerable was exposing the `IArtifactResolutionInfo`